### PR TITLE
Migrate langnames.json

### PR DESCRIPTION
**Pull request by AnonymousWalker**
We integrated [door43-client v0.10.3](https://github.com/unfoldingWord-dev/android-door43-client) library into Writer Android as the catalog URL of `langnames.json` was not configurable in the library above. We change the endpoint and keep the library source code local in our repo.

This also adds a fix to the `truncateTable()` method in `org/unfoldingword/door43client/Library.java` which did not clear the table when executed inside a transaction. See https://github.com/Bible-Translation-Tools/BTT-Writer-Android/pull/79/commits/4938a7189eb3d559c36b448893306afc5477b58d

### CODE REVIEW

**Review by PurpleGuitar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/unfoldingword/door43client/LegacyTools.java [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    /**
+     *
+     * @param library
+     * @param host the host to use for the catalogs
+     * @throws Exception
+     */
+    public static void injectGlobalCatalogs(Library library, String host) throws Exception {
+        host = host != null && !host.trim().isEmpty() ? host : "https://td.unfoldingword.org";
+
+        library.addCatalog(new Catalog("langnames", "https://langnames-temp.walink.org/langnames.json", 0));
```
[the remark this answers is not in the document]
Comment (AnonymousWalker) [reply app/src/main/java/org/unfoldingword/door43client/LegacyTools.java]: Thanks for the suggestion! I added the configuration field in settings. The field is updated when:
- user manually changes the url and saves it
- user switches content server (e.g. WACS to DCS)
- the app is launched

**Review by PurpleGuitar — APPROVED** [review]

**Comment by mXaln** [comment]
> We integrated [door43-client v0.10.3](https://github.com/unfoldingWord-dev/android-door43-client) library into Writer Android as the catalog URL of `langnames.json` was not configurable in the library above. We change the endpoint and keep the library source code local in our repo.
> 
> This also adds a fix to the `truncateTable()` method in `org/unfoldingword/door43client/Library.java` which did not clear the table when executed inside a transaction. See [4938a71](https://github.com/Bible-Translation-Tools/BTT-Writer-Android/commit/4938a7189eb3d559c36b448893306afc5477b58d)

Could we fork this library, update it and then load it in gradle?

**Comment by AnonymousWalker** [comment]
> > We integrated [door43-client v0.10.3](https://github.com/unfoldingWord-dev/android-door43-client) library into Writer Android as the catalog URL of `langnames.json` was not configurable in the library above. We change the endpoint and keep the library source code local in our repo.
> > This also adds a fix to the `truncateTable()` method in `org/unfoldingword/door43client/Library.java` which did not clear the table when executed inside a transaction. See [4938a71](https://github.com/Bible-Translation-Tools/BTT-Writer-Android/commit/4938a7189eb3d559c36b448893306afc5477b58d)
> 
> Could we fork this library, update it and then load it in gradle?

We could. However, this library was built with an outdated version of gradle and some of the dependencies required to build it were no longer available when I attempted to. Furthermore, the library repo appears to be no longer in development for years, which means we just fork and maintain it ourselves. I believe this is why we decided to merge it into BTT Writer Android code base for better maintenance.

**Comment by mXaln** [comment]
> > > We integrated [door43-client v0.10.3](https://github.com/unfoldingWord-dev/android-door43-client) library into Writer Android as the catalog URL of `langnames.json` was not configurable in the library above. We change the endpoint and keep the library source code local in our repo.
> > > This also adds a fix to the `truncateTable()` method in `org/unfoldingword/door43client/Library.java` which did not clear the table when executed inside a transaction. See [4938a71](https://github.com/Bible-Translation-Tools/BTT-Writer-Android/commit/4938a7189eb3d559c36b448893306afc5477b58d)
> > 
> > 
> > Could we fork this library, update it and then load it in gradle?
> 
> We could. However, this library was built with an outdated version of gradle and some of the dependencies required to build it were no longer available when I attempted to. Furthermore, the library repo appears to be no longer in development for years, which means we have to fork and maintain it ourselves. I believe this is why we decided to merge it into BTT Writer Android code base for better maintenance.

All right then

**Review by mXaln — APPROVED** [review]

**Pull request merged**